jenkins/spark-premerge-build.sh

```diff
@@ -108,7 +108,7 @@ ci_2() {
     export TEST_TYPE="pre-commit"
     export TEST_PARALLEL=4
     # separate process to avoid OOM kill
-    TEST='conditionals_test or window_function_test' ./integration_tests/run_pyspark_from_build.sh
+    TEST_PARALLEL=2 TEST='conditionals_test or window_function_test' ./integration_tests/run_pyspark_from_build.sh
```

Comment:
Not sure why this change is part of this PR if GPU OOM during premerge tests are not related to the case when changes?

Reply:
This is to pass the premerge build. 
The OOM we are talking about is not on GPU, but the host memory on CPU.

What I just found is if we add enough new tests of case_when with the default parallel value (it is 4 according to the build log), the ci_2 process will run into OOM (Host memory reached the up limit of a pod).  When reudce it to 2, it can pass.

I doult if we missed to release some memory between tests, or there would be some issues in our premerge build. Since the host memory in the ci_2 process kept increasing during the tests running.